A32NX/html_ui/Pages/A32NX_Utils/NXApi.js

```diff
@@ -14,6 +14,21 @@ class NXApi {
             });
     }
 
+    static getAirport(icao) {
+        if (!icao) {
+            throw ("No ICAO provided");
+        }
+
+        return fetch(`${NXApi.url}/api/v1/airport/${icao}`)
+            .then((response) => {
+                if (!response.ok) {
+                    throw (response);
```

Comment:
Same here, but rather give a useful error message than just throwing the response:

```suggestion
                    throw new Error("Request failed with status code " + response.status);
```